app/src/main/kotlin/at/bitfire/davdroid/repository/AccountRepository.kt

```diff
@@ -218,12 +217,13 @@ class AccountRepository @Inject constructor(
             // update address books
             localAddressBookStore.get().updateAccount(oldAccount, newAccount)
 
-            // calendar provider doesn't allow changing account_name of Events
-            // (all events will have to be downloaded again at next sync)
+            // update calendar events
+            localCalendarStore.get().updateAccount(oldAccount, newAccount)
```

Comment:
There are several possibilities that `localCalendarStore.updateAccount()` throws an exception. Then the rename operation would fail in an undefined state.

I think it would be best to wrap all three `updateAccounts` in the same `catch (Exception)`.